### PR TITLE
Refactor redirect flow and fix URLs for --bind-addr

### PR DESCRIPTION
**Pull request by klizhentas**

### CODE REVIEW

**Comment by osterman** [comment]
Thanks @klizhentas !

File: lib/utils/utils.go [hunk 1]
```diff
@@ -33,6 +36,47 @@ import (
...
+}
+
+// Set sets the value of the string
+func (s *SyncString) Set(v string) {
+	s.Lock()
+	defer s.Unlock()
+	s.string = v
+}
+
+// ClickableURL fixes address in url to m
```
Comment (r0mant) [inline lib/utils/utils.go]: Unfinished comment.

**Review by r0mant — APPROVED** [review]

**Comment by russjones** [comment]
Code looks fine, but it's large enough where it can be encapsulated within it's own struct with methods that you can call from `SSHAgentSSOLogin`.

Something like:

```golang
type callbackServer struct{
}

func (s *callbackServer) handleCallback(w http.ResponseWriter, r *http.Request) {
   [...]
}

func (s *callbackServer) handleRedir(w http.ResponseWriter, r *http.Request) {
   [...]
}

func (s *callbackServer) Start() {
   [...]
}

func (s *callbackServer) Close() error {
   [...]
}
```

**Review by russjones — APPROVED** [review]

**Comment by klizhentas** [comment]
@russjones I'll look into refactor a bit later, may be do some work in master

**Pull request merged**